documentation/tutorials/get-started-with-ash-and-phoenix.md

```diff
@@ -0,0 +1,501 @@
+To use `AshPostgres.Repo` change your repo module to look like this:
+
+```elixir
+# lib/my_ash_phoenix_app/repo.ex
+
+defmodule MyAshPhoenixApp.Repo do
+  use AshPostgres.Repo, otp_app: :my_ash_phoenix_app
+
+  # Installs Postgres extensions that ash commonly uses
+  def installed_extensions do
+    ["uuid-ossp", "citext"]
+  end
+end
+```
+
+After swapping `AshPostgres.Repo` in, you can now create the database using:
+
+```
+$ mix ash_postgres.create
+
+The database for MyAshPhoenixApp.Repo has been created
+```
+
+### Edit Config
+
+We need to specify the Ash APIs that our application uses and some config for backwards compatibility that will be removed in the next major release.
+Add this to your config:
+
+```elixir
+# config/config.exs
+
+import Config
+
+# For backwards compatibility, the following configuration is required.
+# see https://ash-hq.org/docs/guides/ash/latest/get-started#temporary-config for more details
+config :ash, :use_all_identities_in_manage_relationship?, false
+
+config :my_ash_phoenix_app,
+  ash_apis: [MyAshPhoenixApp.Blog]
+```
+
+### Create the API and Registry
+
+An Ash API can be thought of as a [Bounded Context](https://martinfowler.com/bliki/BoundedContext.html) in Domain Driven Design terms and can seen as analogous to a Phoenix context. Put simply, its a way of grouping related resources together. In our case our API is called `MyAshPhoenixApp.Blog`.
+
+An Ash API points to an Ash registry. The registry in our case is `MyAshPhoenixApp.Blog.Registry`
+An Ash registry points to one or more resources. In our case we only have a single resource `MyAshPhoenixApp.Blog.Post`. We'll be taking a deeper look into that in the next section.
+
+For now take a look at the `Blog` API and the `Blog.Registry`:
+
+```elixir
+# lib/my_ash_phoenix_app/blog.ex
+
+defmodule MyAshPhoenixApp.Blog do
+  use Ash.Api
+
+  resources do
+    registry MyAshPhoenixApp.Blog.Registry
+  end
+end
+```
+
+```elixir
+# lib/my_ash_phoenix_app/blog/registry.ex
+
+defmodule MyAshPhoenixApp.Blog.Registry do
+  use Ash.Registry,
+    extensions: [
+      # This extension adds helpful compile time validations
+      Ash.Registry.ResourceValidations
+    ]
+
+  entries do
+    entry MyAshPhoenixApp.Blog.Post
+  end
+end
+```
+
+## Creating Resources
+
+### Creating the `Post` Resource
+
+A resource is a central concept in Ash. In short, a resource is a domain model object in your system. A resource defines the data it holds and defines the actions that can operate on that data.
+
+It's convention to place all the resource in their own resources folder. So when we create `Post` we will place it in `lib/my_ash_phoenix_project/blog/resources/post.ex`. So the structure after making the resource should look like so:
+
+```
+lib/
+├─ my_ash_phoenix_app/
+│  ├─ blog/
+│  │  ├─ registry.ex
+│  │  ├─ resources/
+│  │  │  ├─ post.ex
+│  ├─ blog.ex
+```
+
+Below is the resource module. Read the comments carefully, every line is explained:
+
+```elixir
```

Comment:
It would be good to have the file path here as has been done above.

Reply:
addressed: [675a3a4](https://github.com/ash-project/ash/pull/504/commits/675a3a41730b743fd0bd8b5db419557a0068ff22)